// @vuepress/core/.temp/internal/routes.js
/**
 * Generated by "@vuepress/internal-routes"
 */

import { injectComponentOption, ensureAsyncComponentsLoaded } from '@app/util'
import rootMixins from '@internal/root-mixins'
import GlobalLayout from "/Users/Perhaps/Documents/学习资料/Trae-git/node_modules/@vuepress/core/lib/client/components/GlobalLayout.vue"

injectComponentOption(GlobalLayout, 'mixins', rootMixins)
export const routes = [
  {
    name: "v-75361786",
    path: "/pages/02ce15/",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Layout", "v-75361786").then(next)
    },
  },
  {
    path: "/pages/02ce15/index.html",
    redirect: "/pages/02ce15/"
  },
  {
    path: "/about/",
    redirect: "/pages/02ce15/"
  },
  {
    name: "v-7be604f8",
    path: "/categories/",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Layout", "v-7be604f8").then(next)
    },
  },
  {
    path: "/categories/index.html",
    redirect: "/categories/"
  },
  {
    path: "/@pages/categoriesPage.html",
    redirect: "/categories/"
  },
  {
    name: "v-2393239e",
    path: "/pages/810f02/",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Layout", "v-2393239e").then(next)
    },
  },
  {
    path: "/pages/810f02/index.html",
    redirect: "/pages/810f02/"
  },
  {
    path: "/projects/alpha.html",
    redirect: "/pages/810f02/"
  },
  {
    name: "v-43637224",
    path: "/pages/461738/",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Layout", "v-43637224").then(next)
    },
  },
  {
    path: "/pages/461738/index.html",
    redirect: "/pages/461738/"
  },
  {
    path: "/notes/development.html",
    redirect: "/pages/461738/"
  },
  {
    name: "v-01bad9f8",
    path: "/pages/516fa3/",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Layout", "v-01bad9f8").then(next)
    },
  },
  {
    path: "/pages/516fa3/index.html",
    redirect: "/pages/516fa3/"
  },
  {
    path: "/notes/getting-started.html",
    redirect: "/pages/516fa3/"
  },
  {
    name: "v-ab7250b8",
    path: "/pages/4e2815/",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Layout", "v-ab7250b8").then(next)
    },
  },
  {
    path: "/pages/4e2815/index.html",
    redirect: "/pages/4e2815/"
  },
  {
    path: "/projects/beta.html",
    redirect: "/pages/4e2815/"
  },
  {
    name: "v-3779acb8",
    path: "/archives/",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Layout", "v-3779acb8").then(next)
    },
  },
  {
    path: "/archives/index.html",
    redirect: "/archives/"
  },
  {
    path: "/@pages/archivesPage.html",
    redirect: "/archives/"
  },
  {
    name: "v-27119efd",
    path: "/",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Layout", "v-27119efd").then(next)
    },
  },
  {
    path: "/index.html",
    redirect: "/"
  },
  {
    name: "v-1edc1e24",
    path: "/tags/",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Layout", "v-1edc1e24").then(next)
    },
  },
  {
    path: "/tags/index.html",
    redirect: "/tags/"
  },
  {
    path: "/@pages/tagsPage.html",
    redirect: "/tags/"
  },
  {
    path: '*',
    component: GlobalLayout
  }
]
